edit-post/components/plugin-more-menu-item/index.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { Slot, Fill, withContext, MenuItemsItem } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+
+/**
+ * Name of slot in which the more menu items should fill.
+ *
+ * @type {string}
+ */
+export const SLOT_NAME = 'PluginMoreMenuItem';
+
+let PluginMoreMenuItem = ( { label, onClick, icon, isSelected } ) => (
+	<Fill name={ SLOT_NAME }>
+		{ ( props ) => {
+			return (
+				<MenuItemsItem
+					icon={ isSelected ? 'yes' : icon }
+					isSelected={ isSelected }
+					label={ label }
+					onClick={ () => {
+						onClick();
+						props.onClose();
+					} }
+				/>
+			);
+		} }
+	</Fill>
+);
+
+PluginMoreMenuItem = compose( [
```

Comment:
I still think that it makes more sense to have a separate file with `PluginSidebarMenuItem` which is `switch` free and does all required magic with Sidebar only.